lib/Interpreter/CppInterOp.cpp

```diff
@@ -719,31 +719,46 @@ namespace Cpp {
     return ComputeBaseOffset(getSema().getASTContext(), DCXXRD, Paths.front());
   }
 
-  // FIXME: We should make the std::vector<TCppFunction_t> an out parameter to
-  // avoid copies.
-  std::vector<TCppFunction_t> GetClassMethods(TCppScope_t klass)
-  {
-
-    if (!klass)
-      return {};
-
+  void GetClassMethods(TCppScope_t klass, std::vector<TCppFunction_t> &methods) {
+    if (!klass) return;
+    
     auto *D = (clang::Decl *) klass;
 
     if (auto *TD = dyn_cast<TypedefNameDecl>(D))
       D = GetScopeFromType(TD->getUnderlyingType());
 
-    std::vector<TCppFunction_t> methods;
-    if (auto *CXXRD = dyn_cast_or_null<CXXRecordDecl>(D)) {
+    if (D && isa<CXXRecordDecl>(D)) {
+      auto* CXXRD = dyn_cast<CXXRecordDecl>(D);
+      getSema().ForceDeclarationOfImplicitMembers(CXXRD);
+        for (Decl* DI : CXXRD->decls()) {
+          if (auto* MD = dyn_cast<CXXMethodDecl>(DI))
+            methods.push_back(MD);
+          else if (auto* USD = dyn_cast<UsingShadowDecl>(DI))
+            if (auto* MD = dyn_cast<CXXMethodDecl>(USD->getTargetDecl()))
+              methods.push_back(MD);
```

Comment:
This code and the one in `GetFunctionTemplatedDecls` is very similar please move it out into a separate file static function.

Reply:
done